addons/purchase_stock/models/purchase.py

```diff
@@ -262,13 +262,14 @@ def _compute_qty_received(self):
                             pass
                         elif (
                             move.location_dest_id.usage == "internal"
-                            and move.to_refund
+                            and move.location_id.usage != "supplier"
```

Comment:
not sure about adding this condition in the case of dropship to a subcontractor for example. Is it really needed?

Reply:
The condition makes sure that transfer from Supplier to Internal will increase qty received.

Other part of the change makes sure that is not changed if `to_refund` (*Update quantities on SO/PO*) is not set.

I'm not sure that I understand your example. The source is supplier and destination is internal, right? How should qty be changed in this case?